### DISCUSSION TITLE
Read/write unadvertised characteristics

### DISCUSSION THREAD

**Discussion by furgerf** [capacitor-community/bluetooth-le#629] [Q&A]
Hi, I'd like to read/write characteristics from services that aren't advertised (secondary services).
I have the respective UUIDs but the requests aren't sent, e.g. due to [this guard](https://github.com/capacitor-community/bluetooth-le/blob/main/ios/Plugin/Device.swift#L180).
Is there a way around this?

**Comment by pwespi** [comment]
The services don't have to be advertised, but once connected they have to be visible to service discovery. If you're peripheral changes services at run-time, you can call [`discoverServices`](https://github.com/capacitor-community/bluetooth-le?tab=readme-ov-file#discoverservices) again from your app after connection. To get a list of all discovered services (and their characteristics) you can use [`getServices`](https://github.com/capacitor-community/bluetooth-le?tab=readme-ov-file#getservices).

**Reply by furgerf** [reply]
Thanks for your reply. I'm afraid I don't quite follow though, isn't a service only visible to service discovery when it's advertised? What distinction are you referring to?
